src/main/java/org/jlab/mya/event/EventCode.java

```diff
@@ -11,60 +15,72 @@ public enum EventCode {
     /**
      * Update
      */
-    UPDATE(0, "Normal channel data point", false),
+    UPDATE(0, "Normal channel data point"),
     /**
      * Network Disconnection
      */
-    NETWORK_DISCONNECTION(1, "Network disconnection", true),
+    NETWORK_DISCONNECTION(1, "Network disconnection"),
     /**
      * Archiving Disabled
      */
-    ARCHIVING_OF_CHANNEL_TURNED_OFF(2, "Archiving of channel turned off", true),
+    ARCHIVING_OF_CHANNEL_TURNED_OFF(2, "Archiving of channel turned off"),
     /**
      * Archiver Shutdown
      */
-    ARCHIVER_SHUTDOWN(3, "Archiver shutdown", true),
+    ARCHIVER_SHUTDOWN(3, "Archiver shutdown"),
     /**
      * Unknown
      */
-    UNKNOWN_UNAVAILABILTY(4, "Unknown unavailability", true),
+    UNKNOWN_UNAVAILABILTY(4, "Unknown unavailability"),
     /**
      * NaN or Infinity
      */
-    NAN_OR_INFINITY(5, "NaN/infinity encountered", false),
+    NAN_OR_INFINITY(5, "NaN/infinity encountered"),
     /**
      * Origin
      */
-    ORIGIN_OF_CHANNELS_HISTORY(16, "Origin of channel's history", false),
+    ORIGIN_OF_CHANNELS_HISTORY(16, "Origin of channel's history"),
     /**
      * Offline
      */
-    CHANNELS_PRIOR_DATA_MOVED_OFFLINE(32, "Channel's prior data moved offline", false),
+    CHANNELS_PRIOR_DATA_MOVED_OFFLINE(32, "Channel's prior data moved offline"),
     /**
      * Discarded
      */
-    CHANNELS_PRIOR_DATA_DISCARDED(48, "Channel's prior data discarded", false),
+    CHANNELS_PRIOR_DATA_DISCARDED(48, "Channel's prior data discarded"),
     /**
-     * Undefined
+     * Undefined.  This event code is not defined in the C++ API.  It is used to indicate the lack of the known state of
+     * the channel and is an "artificial" event.  This is best used to simulate an event that happens before the start
+     * of recorded data in the database or an event that happens in the future.  mySamplerStream provides a good example
+     * of this.
      */
-    UNDEFINED(128, "undefined", false);
+    UNDEFINED(255, "undefined");
 
     private final int codeNumber;
     private final String description;
     private final boolean disconnection;
 
+    private static final Set<EventCode> dataEventCodes = Collections.unmodifiableSet(Stream.of(UPDATE,
```

Comment:
Looks like there are two ways to determine if an event has a value: use isDisconnection() method or use getDataEventCodes() method.   Are they supposed to agree?  Should we verify they always agree?  Looking at old code it appears NAN was (possibly erroneously) categorized as a data event whereas getDataEventCodes() says otherwise.

Reply:
The NAN event code is 5 or 0x05 in hex.  Given the low nibble is non-zero it's meant to be treated as a disconnect event.  I believe it was treated wrong in earlier versions.  It's a little confusing since NaN is a valid floating number, but these results are typically generated by 3rd party CA servers that were having problems.  I believe the choice was made upstream to treat it as a disconnect since there was no valid data coming from the CA server.  I've reached out for confirmation on this.

The isDisconnection() checks an individual event, while the getDataEvent is meant to make it easy to track down values in the database when someone request "data only" streams.  It's probably a good idea to add a unit test to make sure we don't get these out of sync.  I'll work on that.

Reply:
I've added some tests.  One verifies that all EventCodes are exclusively either disconnect or data events.  Others try to verify that the code I updated from "!= EventCode.Update" checks to ".isDisconnection()" checks treats the no prior data EventCodes the same as UPDATE events.  Some of those tests end up a little artificial since no prior data events should only ever be the first event in a stream, but I wasn't sure how to test some behavior without putting them in the middle of the stream.

Reply:
I got confirmation that NaN is intended as a disconnect event and that no NaN values will be present as MYA data.